src/assets/githubAction/githubActionBeginner.md

```diff
@@ -0,0 +1,129 @@
+authorAvatar : https://avatars.githubusercontent.com/u/61624650?s=400&u=8f992a9ca189af19c40cb070a5af1e02669d3884&v=4
+authorLink: https://github.com/nabim777
+createdAt: Oct 30, 2023
+tags: CI, CD, github, githubAction
+banner: https://blog.jankaritech.com/src/assets/githubAction/images/githubaction_banner.png
+---
+GitHub Actions is an automation and CI/CD service provided by GitHub, allowing us to automate our software development processes or workflows, from building and testing code to deploying it to different environments. It's an effective tool that can enhance software quality and save time.
+
+## Some features of GitHub Actions:
+- It allows you to automate your Software Development Life Cycle (SDLC) processes or workflows.
+- It simplifies CI/CD integration, facilitating seamless DevOps practices.
+
+## Components of GitHub Actions:
+1. Workflows
+2. Events
+3. Jobs
+4. Steps
+5. Actions
+6. Runners
+
+![components](/src/assets/githubAction/images/components_of_github_action.png)
+
+
+### 1. Workflows:
+Workflows are automated procedures that you add to your repository. It will run when triggered by an event in your repository, or they can be triggered manually, or at a defined schedule. It can be used to build, test, package, release, or deploy a project on GitHub.
+
+### 2. Events:
+Events are specific activities such as code push, pull request creation, issue creation, comments on pull requests or issues, etc., that trigger a workflow.
+
+### 3. Jobs
+Jobs are sets of steps that are executed in the same runner. A workflow with multiple jobs will run in parallel, which is default, but it can be configured so that jobs run sequentially.
+
+### 4. Steps
+A step is an individual task that can run the commands in a job. These can be actions or commands.
+
+### 5. Actions
+Actions are the standalone commands for the GitHub Actions platform that perform complex but frequently repeated tasks. Actions help to reduce repetitive code in workflows.
+
+### 6. Runner
+A runner is a GitHub Actions server where our workflows run. It listens for available jobs, runs multiple jobs at a time, and reports the progress, logs, and results back to GitHub. GitHub hosted runners are based on Ubuntu, Linux, Microsoft Windows, and Mac OS.
+
+Note:
+One event can trigger many workflows, a workflow can contain many jobs, and a job can contain many steps.
+
+### Process in GitHub Actions:
+
+![process_in_githubActions](/src/assets/githubAction/images/process.png)
+
+## Getting started:
+
+Before we dive into an example, make sure you have a repository where you want to set up your CI/CD workflow. If you don't have one, create a new repository, for example, named `action-hero`.
+
+Step 1: Clone your repository into your system
+```bash
+git clone <your_github_repo_url>
+```
+
+Step 2: Go to path `action-hero` and create a folder named `.github`
+```bash
+cd action-hero
+mkdir .github
+```
+
+Step 3: Then, go to path `.github` and create a folder named `workflows`
```

Comment:
```suggestion
Step 3: Go to path `.github` and create a folder named `workflows`
```
"Then" is not needed since it's the only thing your telling the reader to do in this step